### ISSUE TITLE
Cleaning progress output

### ISSUE THREAD

**Issue by 4g0tt3nSou1** [oSumAtrIX/DownOnSpot#95]
**Is your feature request related to a problem? Please describe.**
Some CLI-based apps, similar to this, will need to update terminal with progress of downloads, etc. Some do like the app currently does where it just outputs everything again with the changes

**Describe the solution you'd like**
Rather than hitting the terminal with updates on ``"refresh_ui_seconds"``, update the status in-line similar to https://cli.r-lib.org/articles/progress-advanced.html

Basically, it seems cumbersome to see progress of larger lists when it continues to post everything again to update the counter, Sure you can change the interval to allow more readability, but it'd be slightly cleaner to simply show what it's doing with a progress counter in same output like that link, and maybe an overall progress bar showing the percentage of task left

**Additional context**
**_VERY_** minor feature, just basically asking for cleaner output.
